src/main.cpp

```diff
@@ -1,151 +1,119 @@
+        }
+    }
 
-    // dma config
-    tts::dma_setup();
+    // f - fall
+    if (key_input == 102) {
+        if (shoot::throttle_code <= MAX_THROTTLE &&
+            shoot::throttle_code >= ZERO_THROTTLE) {
+            if (shoot::throttle_code == ZERO_THROTTLE) {
+                printf("Throttle is zero\n");
+            } else {
+                shoot::throttle_code -= 50;
+                printf("Throttle: %i\n", shoot::throttle_code - ZERO_THROTTLE);
+            }
+        } else {
+            printf("Motor is not in throttle mode\n");
+        }
+    }
 
-    // Set repeating timer
-    // NOTE: this can be put in main loop to start 
-    // repeating timer on key press (e.g. a for arm)
-    shoot::rt_setup();
+    // q - disarm
+    // Not sure how to disarm yet. Maybe set throttle to 0 and don't send a
+    // cmd for some secs?
+    if (key_input == 32) {
+        shoot::throttle_code = ZERO_THROTTLE;
+        shoot::telemetry = 0;
+        printf("Throttle: %i\n", 0);
+    }
 
-    delay(1500);
+    shoot::send_dshot_frame();
+    printf("Finished processing byte.\n");
+}
 
-    tts::print_gpio_setup();
-    tts::print_dshot_setup();
-    tts::print_pwm_setup();
-    tts::print_dma_setup();
+int main() {
+    int32_t thrust;
+    int key_input;
 
-    Serial.print("Initial throttle: ");
-    Serial.print(shoot::throttle_code);
-    Serial.print("\tInitial telemetry: ");
-    Serial.println(shoot::telemetry);
-}
+    // Load cell calibration
+    int32_t zero_val = 150622;
 
-int incomingByte;
+    stdio_init_all();
+    gpio_init(LED_BUILTIN);
+    gpio_set_dir(LED_BUILTIN, GPIO_OUT);
 
-void loop()
-{
-    if (Serial.available() > 0)
-    {
-        incomingByte = Serial.read();
-        Serial.print("Byte: ");
-        Serial.println(incomingByte, DEC);
+    for (int i = 0; i < 3; i++) {
+        gpio_put(LED_BUILTIN, 1);  // Set pin 25 to high
+        printf("LED ON!\n");
+        sleep_ms(1000);  // 0.5s delay
 
-        // l (led)
-        if (incomingByte == 108)
-        {
-            utils::flash_led(LED_BUILTIN);
-        }
+        gpio_put(LED_BUILTIN, 0);  // Set pin 25 to low
+        printf("LED OFF!\n");
+        sleep_ms(1000);  // 0.5s delay
+    }
 
-        // a (arm)
-        // if (incomingByte == 97)
-        // {
-        //     // Disable timer
-        //     cancel_repeating_timer(&send_frame_rt);
-        //     // Arm motor
-        //     arm_motor();
-        //     // Re-enable timer
-        //     dma_alarm_rt_state = alarm_pool_add_repeating_timer_us(pico_alarm_pool, DMA_ALARM_PERIOD, repeating_send_dshot_frame, NULL, &send_frame_rt);
-        //     Serial.print("Re-enabled repeating DMA alarm: ");
-        //     Serial.println(dma_alarm_rt_state);
-
-        //     return;
-        // }
-
-        // b - beep
-        if (incomingByte == 98)
-        {
-            shoot::throttle_code = 1;
-            shoot::telemetry = 1;
-        }
+    printf("Setting up load cell\n");
```

Comment:
Ideally this should be wrapped in a function and called in main.

Reply:
Do you mean like a "setup" function for the load cells?

Reply:
Yes pls

Reply:
This is a bit more complicated than I thought as the hx711 library has its own init functions. Will look into this further.

Reply:
Leaving this to be implemented in a later date as it is not so important for this PR